### PR TITLE
[X86][APX] Remove KEYLOCKER and SHA promotions from EVEX MAP4.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
APX spec: https://cdrdv2.intel.com/v1/dl/getContent/784266
Change happended in version 4.0.
Removed instructions' Opcodes:
AESDEC128KL
AESDEC256KL
AESDECWIDE128KL
AESDECWIDE256KL
AESENC128KL
AESENC256KL
AESENCWIDE128KL
AESENCWIDE256KL
ENCODEKEY128
ENCODEKEY256
SHA1MSG1
SHA1MSG2
SHA1NEXTE
SHA1RNDS4
SHA256MSG1
SHA256MSG2
SHA256RNDS2

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrSSE.td [hunk 1]
```diff
@@ -6740,55 +6740,19 @@ let Constraints = "$src1 = $dst", Predicates = [HasSHA, NoEGPR] in {
...
-                     EVEX, NoCD8, T_MAP4;
-
-  defm SHA256MSG1 : SHAI_binop<0xDC, "sha256msg1", int_x86_sha256msg1,
-                               SchedWriteVecIMul.XMM, "_EVEX">,
-                    EVEX, NoCD8, T_MAP4;
-  defm SHA256MSG2 : SHAI_binop<0xDD, "sha256msg2", int_x86_sha256msg2,
-                               SchedWriteVecIMul.XMM, "_EVEX">,
-                    EVEX, NoCD8, T_MAP4;
-}
+// Aliases with explicit %xmm0
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrSSE.td:6751]: Put it in Alias.td

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrSSE.td]: [ec540b6](https://github.com/llvm/llvm-project/pull/89173/commits/ec540b6f4cea6a9df180d4dcd5b54da1ec4b149f)

**Comment by KanRobert** [comment]
Could you list the opcodes for your change in the description? I noticed AESENCWIDE256KL is removed in your patch, is it expected?

**Comment by FreddyLeaf** [comment]
> Could you list the opcodes for your change in the description? I noticed AESENCWIDE256KL is removed in your patch, is it expected?

Done. it is expected. See version 4.0

File: llvm/lib/Target/X86/X86ISelDAGToDAG.cpp [hunk 2]
```diff
@@ -5047,17 +5047,17 @@ void X86DAGToDAGISel::Select(SDNode *Node) {
...
-        Opcode = GET_EGPR_IF_ENABLED(X86::ENCODEKEY256);
+        Opcode = X86::ENCODEKEY256;
         break;
       }
 
       SDValue Chain = Node->getOperand(0);
       Chain = CurDAG->getCopyToReg(Chain, dl, X86::XMM0, Node->getOperand(3),
                                    SDValue());
-      if (Opcode == X86::ENCODEKEY256 || Opcode == X86::ENCODEKEY256_EVEX)
+      if (Opcode == X86::ENCODEKEY256)
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86ISelDAGToDAG.cpp:5060]: This opcode is not mentioned in the description

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86ISelDAGToDAG.cpp:5060]: Sorry, updated.

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**